### ISSUE TITLE
Running process on shared .NET 6 runtime encounters "could not load assembly" error after Windows Update

### ISSUE THREAD

**Issue by logicaloud** [dotnet/runtime#93995]
We have a running Windows service process with reported issues on Windows 11 and possibly Windows Server 2022. The process uses the .NET 6 shared framework. After a Windows Update, the process reported that assembly "System.Security.Cryptography.Algorithms, Version 6.0.0.0" could not be loaded. Restarting the service solved the problem.

The installation folder of the Windows service does not contain the "System.Security.Cryptography.Algorithms" assembly but contains some other "System.Security.*" related assemblies that may depend on "System.Security.Cryptography.Algorithms". Assemblies are not dynamically loaded.

The error was encountered after some user interaction with the service that caused it to attempt some cryptography related operation. I doubt that this is a cryptography related problem though; presumably the Windows Update simply touched that particular dependency and it may well be that the error can be encountered loading a different assembly.

It appears that, after the Windows update, there could be a latent issue in a running service, and the impact of that issue can not be foreseen.

What I am trying to understand is:

1. Should this issue appear in the first place? The running service does not change dependencies and yet it can no longer load the assembly.
2. If there is a Windows Update and there are running processes that depend on some components about to be updated, should Windows prompt the user to reboot?
3. Is there a way, possibly in the .NET runtime, to detect a change in the dependencies that are broken or that a Windows Update has occurred that may impact on its operation?

In short, how can I know that the running service can provide all the features that are built into it after a Windows Update?

**Comment by moellerm** [comment]
I suspect your problem is issue #60144.

Windows update and the .NET runtime installers do an "in-place" upgrade for Servicing updates (for example from 6.0.22 to 6.0.24). But the "in place" upgrade is not done in place, but moves it to a different path, ending with the new version number. This leaves running processes with just a partial old runtime (the files that are open when the upgrade happens). Any other dll can't be found on the path of the old runtime, because it does not exist anymore.

I think the installer should either do a real "in place" upgrade where for example 6.0.22 and 6.0.24 both are placed in a path ending in ./6.0 or keep the old runtime untouched until the system is rebooted. That would mean, install 6.0.24 side by side with 6.0.22 and schedule an uninstall for 6.0.22 for the next reboot.

**Comment by logicaloud** [comment]
Ah, thanks. I have missed that one. Closing this as duplicate.
